docs/webdriver-bidi.md

```diff
@@ -0,0 +1,187 @@
+
+Example of launching Firefox with WebDriver BiDi:
+
+```ts
+import puppeteer from 'puppeteer';
+
+const browser = await puppeteer.launch({
+  product: 'firefox',
+  protocol: 'webDriverBiDi',
+});
+const page = await browser.newPage();
+...
+await browser.close();
+```
+
+Example of launching Chrome with WebDriver BiDi:
+
+```ts
+import puppeteer from 'puppeteer';
+
+const browser = await puppeteer.launch({
+  product: 'chrome',
+  protocol: 'webDriverBiDi',
+});
+const page = await browser.newPage();
+...
+await browser.close();
+```
+
+## Puppeteer features supported over WebDriver BiDi
+
+- Browser and page automation
+
+  - Puppeteer.launch
+  - Browser.close
+  - Page.bringToFront
+  - Page.goto
+  - Page.reload (except for `ignoreCache` parameter)
+  - Page.setViewport (`width`, `height`, `deviceScaleFactor` only)
+
+- [Script evaluation](https://pptr.dev/guides/evaluate-javascript):
+
+  - Page.evaluate
+  - Page.exposeFunction
+  - JSHandle.evaluate
+  - JSHandle.evaluateHandle
+
+- [Selectors](https://pptr.dev/guides/query-selectors) and [locators](https://pptr.dev/guides/locators) except for ARIA:
+
+  - Page.$ (ARIA selectors only supported in Chrome)
+  - Page.$$ (ARIA selectors only supported in Chrome)
+  - Page.$eval (ARIA selectors only supported in Chrome)
+  - Page.$$eval (ARIA selectors only supported in Chrome)
+  - Frame.goto() (except `referer` and `referrerPolicy`)
+  - Page.waitForSelector (ARIA selectors only supported in Chrome)
+
+- Input
+
+  - ElementHandle.click
+  - Keyboard.down
+  - Keyboard.press
+  - Keyboard.sendCharacter
+  - Keyboard.type
+  - Keyboard.up
+  - Mouse events (except for dedicated drag'n'drop API methods)
+  - TouchScreen.\*
+  - Page.tap
+
+- JavaScript dialog interception
+
+  - page.on('dialog')
+  - Dialog.\*
+
+- Screenshots (not all parameters are supported)
+
+  - Page.screenshot (supported parameters `clip`, `encoding`, `fullPage`)
+
+- PDF generation (not all parameters are supported)
+
+  - Page.pdf (only `format`, `height`, `landscape`, `margin`, `pageRanges`, `printBackground`, `scale`, `width` are supported)
+  - Page.createPDFStream (only `format`, `height`, `landscape`, `margin`, `pageRanges`, `printBackground`, `scale`, `width` are supported)
+
+## Puppeteer features not yet supported over WebDriver BiDi
+
+- [Request interception](https://pptr.dev/guides/request-interception)
+
+  - HTTPRequest.client()
+  - HTTPRequest.continueRequestOverrides()
+  - HTTPRequest.responseForRequest()
+  - HTTPRequest.abortErrorReason()
+  - HTTPRequest.interceptResolutionState()
+  - HTTPRequest.isInterceptResolutionHandled()
+  - HTTPRequest.finalizeInterceptions()
+  - HTTPRequest.failure()
+  - HTTPRequest.continue()
+  - HTTPRequest.respond()
+  - HTTPRequest.abort()
+  - Page.setRequestInterception()
+
+- Permissions
+
+  - BrowserContext.overridePermissions()
+  - BrowserContext.clearPermissionOverrides()
+
+- Various emulations (most are only supported with Chrome)
```

Comment:
```suggestion
- Various emulations (most are supported with Chrome)
```

Reply:
This should be a little bit better as we tell - These features are unsupported by BiDi and here we inverse it to Most emulations are supported by BiDi for Chrome.